bcgov/healthgateway
# Create public covid test pages for AB#11788.

# Fixes or Implements [AB#11788](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/11788)


## Description

Created public covid test input form and result pages as per requirements in [AB#11701](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/11701).

1. Get your COVID-19 test result input form page
2. Your COVID-19 test result page
3. On Your COVID-19 test result page, 'Positive' is shown with 'text-danger' class and 'Negative' is snow with 'text-success' class
4. Result link opens in new tab on page
5. Mismatch error is shown when statusIndicator is "NotFound" or "DataMismatch" 
6. Our Apologies..... is show when statusIndicator is "Threshold" or "Blocked"
7. Add snow plow call
8. Update home link to deal with store errors - British Columbia Logo top left corner
9. Addresses click vs blur issue on cancel and home link. Using mouse down will execute first over blur. As a result, validation on PHN input field will not fire. If mouse down is not used, then you may have to click on cancel button twice.
10. Addresses cancel button issue if errors exist in store. If an error occurs via store and user cancels out or clicks on home link and then subsequently navigates back to covid test input form, errors will not display.  Errors will display if store object is not re-initialized.

See screenshots:

**Input page form:**
<img width="2035" alt="AB-11788-FORM" src="https://user-images.githubusercontent.com/58790456/143941115-754a61a9-da4e-4753-9853-beb9368a7455.png">

**Result page with 2 records:**
<img width="1022" alt="AB-11788-2-RECORDS" src="https://user-images.githubusercontent.com/58790456/143941213-cdc5b4a6-18f9-4fa3-82b6-8f2fcd0b8826.png">

**Result page with 1 record**
<img width="2042" alt="AB-11788-1-RECORD" src="https://user-images.githubusercontent.com/58790456/143941250-9803b564-042f-4c39-8cc7-178501578a55.png">

**Result page with some result items missing**
<img width="2032" alt="AB-11788-MISSING-DATA" src="https://user-images.githubusercontent.com/58790456/143941362-19c073d2-ba92-4054-af13-903d17c49441.png">

**Result page with Mismatch Error**
<img width="2045" alt="AB-11788-MISMATCH" src="https://user-images.githubusercontent.com/58790456/143941437-8a6bc149-3a30-4f7e-9905-6cd579a35b18.png">

**Result page with Apologies.... Error**
<img width="1020" alt="AB-11788-INVALID" src="https://user-images.githubusercontent.com/58790456/143941562-ed0ac949-6199-4389-bd06-5a23b0a8daf1.png">

**Results page - iPhone 5**
<img width="561" alt="AB-11788-IPHONE-5" src="https://user-images.githubusercontent.com/58790456/143941614-d1854195-3a72-4ddb-922e-1472282be3e9.png">

**Results page - iPhone 6**
<img width="562" alt="AB-11788-IPHONE-6" src="https://user-images.githubusercontent.com/58790456/143941672-02a47164-d7d3-4834-b381-26b96363cc04.png">

**Results page - Galaxy S5**
<img width="515" alt="AB-11788-GALAXY-S5" src="https://user-images.githubusercontent.com/58790456/143941748-a5558ea5-261e-40a6-ad62-0cb8999b0a62.png">
If you are reviewing this PR, please adhere to the guidelines as [documented](https://github.com/bcgov/healthgateway/wiki/prguidance).

**Results page - No results from PHSA**
<img width="2012" alt="AB-11788-NO-RESULTS" src="https://user-images.githubusercontent.com/58790456/143947703-d77e1844-2be3-4c90-939b-edb6aaf7af7b.png">
## Testing

-   [ ] Functional Tests Updated

### Steps to Reproduce

If this is a bug, please provide details on how to reproduce the code.

### UI Changes

YES

### Browsers Tested


Provide an explanation for any test(s) not completed.

## Notes/Additional Comments

Any additional notes or comments that may be relevant. Include any specialized deployment steps here.

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

-   Fulfills Work Item Requirements

    -   The changes meet/implement story's acceptance criteria. Fixes identified problems if bug.
    -   Changes not directly related to the task requirements need to be documented on the PR.

-   Compilation / Tests

    -   The changes do not break compilation and/or unit or functional tests; unless other item addresses them specifically.

-   Logic Problems / Functional Behaviour

    -   The changes work as intended and do not introduce problems.

-   Performance

    -   The changes do not introduce obvious performance issues.

-   Documented Standards

    -   The changes adhere to the team's documented [Coding Standards](https://github.com/bcgov/healthgateway/wiki/standards).

-   Readability / Maintainability
    -   The changes can be easily understood/read and allow for future enhancements without major refactoring. Readability is preferred over "clever code".
    -   Reasoning for changes needs to be clearly articulated. Disagreements should be arbitrated by a third developer.
